File: wwwroot/ts/DTOs/ImageDataDTO.ts
export default class ImageDataDTO {
    canvasWidth: number
    canvasHeight: number
    encodedImage: string
    imageName: string
    frameNumber: number
    fileExtension: string

    constructor(canvasWidth: number, canvasHeight: number, encodedImage: string,
                imageName: string, frameNumber: number, fileExtension?: string) {
        this.canvasWidth = canvasWidth
        this.canvasHeight = canvasHeight
        this.encodedImage = encodedImage

        this.imageName = imageName
        this.frameNumber = frameNumber

        this.fileExtension = fileExtension ?? "jpeg"; // Default if not provided
    }
}

File: Flipbook_App.Server/Models/DTOs/ImageDataDTO.cs
namespace Flipbook_App.Models.DTOs;

public class ImageDataDTO
{
	public int CanvasWidth { get; set; }

	public int CanvasHeight { get; set; }

	public string EncodedImage { get; set; }

	public string ImageName { get; set; }

	public string FileExtension { get; set; }

	public int FrameNumber { get; set; }
}
